TAILQ kicks in.

0mp opened the pull request.

0mp committed:
Remove assert.h from bsmconv.c.

0mp committed:
Update and rename test script.

0mp committed:
Add a successfully initialized TAILQ.

More queues soon.

0mp committed:
Rename the fu script.

0mp committed:
Add extracting of timestamp and id from a record.

0mp committed:
Add tests and modify the fu script.

0mp committed:
Parse timestamp and id. Clean up before parsing fields.

Some tests are still failling.

0mp committed:
Fix the sum function of secs and nsecs.

0mp committed:
Fix test files. (Used ':' instead of '=' in a record's fields).

0mp committed:
Parse record fields. (In progress.)

- Perform s/pjdlog_notice/pjdlog_debug/g.

0mp committed:
Fix pjdlog_debug usage.

0mp committed:
Remove commas from tests. Add a negative comma test.

There should be no commas between records' fields for the time being.

0mp committed:
Improve debug logging. It is possible to control debug level now.

As you normally run `fu t` to run tests you can now run
`fu tv -vvv` to run tests with the debug level of 3.

0mp committed:
Fix fu, the test script.

0mp committed:
Remove unnecessary defines.

0mp committed:
Fix the progress bar in fu, the test script.

0mp committed:
Fix fu, the testing script.

0mp committed:
Fix the sumsecs equation.

0mp committed:
Update and add some tests.

kwitaszczyk commented on contrib/openbsm/bin/bsmconv/bsmconv.c:
```diff
@@ -78,111 +118,303 @@ find_msg_field_end(struct sbuf *buf, const size_t pos)
 }
 
-int main()
+int main(int argc, char *argv[])
```
Style: return type should be in a separate line.

kwitaszczyk commented on contrib/openbsm/bin/bsmconv/bsmconv.c:
```diff
@@ -191,14 +423,12 @@ int main()
-	eventbuf = sbuf_new_auto();
-	PJDLOG_ASSERT(eventbuf != NULL);
+	int debuglevel;
 
```
Style: unnecessary blank line.

0mp replied:
Solved.

0mp committed:
Solve #21 (the issue with errno != 0).

0mp committed:
Add a test and update the test script.

0mp committed:
Update tests.

0mp committed:
Update tests.

0mp committed:
Parse fields of a record.

I've added a lot of minor changes apart from that.

0mp committed:
Add linau_record_init and solve some TODOs.

0mp committed:
Change mallocs to callocs.

- Incorporate the PJDLOG_VERIFY function.
- Review the correctness of the usage of the PJDLOG_ASSERT function.
  Since asserts can be turned off so use PJDLOG_VERIFY instead of
  PJDLOG_ASSERT if you call any significant functions within the macro.

0mp committed:
Use PJDLOG_VERIFY instead of PJDLOG_ASSERT.

Previously, I checked (bytesread != 1) using an assert which might
result in an uncaught read(2) error.

0mp committed:
Fix minor issues.

0mp committed:
Update string_to_unit32().

Also:
- Change some debug messages.
- Reenable the negative test set.
